### PR TITLE
extended commentary on waf disable

### PR DESCRIPTION
**Pull request by electricjesus**
Added some needed commentary on how the Application Layer CR works

Product Version(s):

Calico Enterprise, Calico Cloud

Issue:

https://tigera.atlassian.net/browse/RS-1440

Link to docs preview:

SME review:
- [x] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 1]
```diff
@@ -318,15 +318,34 @@ kubectl annotate svc <service-name> -n <service-namespace> projectcalico.org/l7-
 
 ### Disable WAF feature from your cluster
 
-To disable WAF, update the [ApplicationLayer](../reference/installation/api#operator.tigera.io/v1.ApplicationLayer) resource to include the `webApplicationFirewall` field, and ensure it is set to `Disabled`.
+To disable WAF safely, update the [ApplicationLayer](../reference/installation/api#operator.tigera.io/v1.ApplicationLayer) accordingly:
 
-Example:
+#### If you have more than one ApplicationLayer feature enabled:
 
+[ApplicationLayer Specification](../reference/installation/api#operator.tigera.io/v1.ApplicationLayerSpec) can specify configuration for [Application Logging](../reference/installation/api#operator.tigera.io/v1.LogCollectionSpec) and [Application Layer Policy](../reference/installation/api#operator.tigera.io/v1.ApplicationLayerPolicyStatusType) as well. However, for this custom resource to be valid, at least one of these features have to be enabled e.g.:
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:325]: ### 
```suggestion
The [ApplicationLayer Specification](../reference/installation/api#operator.tigera.io/v1.ApplicationLayerSpec) can specify configuration for [Application Logging](../reference/installation/api#operator.tigera.io/v1.LogCollectionSpec) and [Application Layer Policy](../reference/installation/api#operator.tigera.io/v1.ApplicationLayerPolicyStatusType) as well. However, for this custom resource to be valid, at least one of these features must be enabled, for example:
```

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 2]
```diff
@@ -318,15 +318,34 @@ kubectl annotate svc <service-name> -n <service-namespace> projectcalico.org/l7-
...
+
+Example: WAF enabled, ALP disabled -- this is valid
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:327]:
```suggestion
Valid example: WAF enabled, ALP disabled 
```

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 3]
```diff
@@ -318,15 +318,34 @@ kubectl annotate svc <service-name> -n <service-namespace> projectcalico.org/l7-
...
+apiVersion: operator.tigera.io/v1
+kind: ApplicationLayer
+metadata:
+  name: tigera-secure
+spec:
+  webApplicationFirewall: Disabled
+  applicationLayerPolicy: Enabled
+```
+
+Example: WAF and ALP both set as disabled, log collection is omitted but the default is disabled -- this is invalid
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:338]:
```suggestion
Invalid example: WAF and ALP are both set as disabled, and log collection is omitted but the default is disabled
```

File: calico-enterprise/threat/web-application-firewall.mdx [hunk 4]
```diff
@@ -318,15 +318,34 @@ kubectl annotate svc <service-name> -n <service-namespace> projectcalico.org/l7-
 
 ### Disable WAF feature from your cluster
```
Comment (bmckercher123) [inline calico-enterprise/threat/web-application-firewall.mdx:348]: ### Disable WAF feature from your cluster

To safely disable WAF, determine how to handle ApplicationLayer features and follow the steps:

- **Disable all ApplicationLayer features**

   You must delete the ApplcationLayer CR `tigera-secure`, for example:

    `kubectl delete applicationlayer tigera-secure`


- **Keep some ApplicationLayer features enabled**

To disable WAF but keep some ApplicationLayer features enabled, you must update the [ApplicationLayer](../reference/installation/api#operator.tigera.io/v1.ApplicationLayer). Note that the [ApplicationLayer Specification](../reference/installation/api#operator.tigera.io/v1.ApplicationLayerSpec) can specify configuration for both [Application Logging](../reference/installation/api#operator.tigera.io/v1.LogCollectionSpec) and [Application Layer Policy](../reference/installation/api#operator.tigera.io/v1.ApplicationLayerPolicyStatusType). 

For the ApplicationLayer custom resource to be valid, at least one of these features must be enabled, for example:

**Valid YAML**

WAF is enabled, and is ALP disabled.

```yaml
apiVersion: operator.tigera.io/v1
kind: ApplicationLayer
metadata:
  name: tigera-secure
spec:
  webApplicationFirewall: Disabled
  applicationLayerPolicy: Enabled
```

**Invalid YAML**

WAF and ALP are disabled, and log collection is omitted (and the default is disabled). 

```yaml
apiVersion: operator.tigera.io/v1
kind: ApplicationLayer
metadata:
  name: tigera-secure
spec:
  webApplicationFirewall: Disabled
  applicationLayerPolicy: Disabled
```

**Review by bmckercher123 — CHANGES_REQUESTED** [review]
I had trouble following the choices. Suggestions for clarity (if my interpretation is correct).

**Review by bmckercher123 — APPROVED** [review]
An error on my part....delete the comma after WAF in this sentence. Otherwise, LTGM and go ahead and copy. 

To disable WAF, but

**Review by bmckercher123 — APPROVED** [review]
LGTM. Do a rebase and I'll merge it.

**Comment by bmckercher123** [comment]
@electricjesus You'll need to fix this link to build successfully. This link (https://github.com/tigera/docs/reference/installation/api#operator.tigera.io/v1.ApplicationLayer), should be an internal link like the others: /..reference/installation/api#operator.tigera.io/v1

**Comment by ctauchen** [comment]
@electricjesus Checking in on this. Looks like you were mostly there. Are you still hoping to merge this? Needs picks to new versions we've cut since you last worked on this.

**Comment by ctauchen** [comment]
@electricjesus Closing this for now, please reopen if you want to revisit this material.

**Comment by peterkellydev** [comment]
This is definitely needed. I am reopening this to get it over the line.

**Comment by ctauchen** [comment]
@peterkellydev Sure thing. @electricjesus I think all that's needed is:

* add to new versions 
* rebase, sort conflicts

After that we should be good to go.

**Comment by ctauchen** [comment]
LGTM

**Pull request merged**